src/vm/assembly.cpp

```diff
@@ -1983,6 +2010,58 @@ BOOL Assembly::IsInstrumentedHelper()
 }
 #endif // FEATURE_PREJIT
 
+
+#ifdef FEATURE_COMINTEROP
+
+ITypeLib *Assembly::InvalidTypeLib = (ITypeLib *)-1;
+
+ITypeLib* Assembly::GetTypeLib()
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        FORBID_FAULT;
+    }
+    CONTRACTL_END
+
+    ITypeLib *pTlb = m_pITypeLib;
+    if (pTlb != nullptr && pTlb != Assembly::InvalidTypeLib)
+        pTlb->AddRef();
+
+    return pTlb;
+} // ITypeLib* Assembly::GetTypeLib()
+
+void Assembly::SetTypeLib(ITypeLib *pNew)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        FORBID_FAULT;
+    }
+    CONTRACTL_END
+
+    ITypeLib *pOld;
```

Comment:
I do not think that this is correct - it has the same issue as the code that was here before.

The only valid transitions of the field should be: `null -> -1` and `null -> non-null`.

Once the field is non-null, it should never change its value.

Reply:
That is handled in the caller side. Are you indicating you want the semantics to be enforced from within the assembly class?